File: backend_kapda_junction/src/models/Banner.js
const mongoose = require('mongoose');

const bannerSchema = new mongoose.Schema({
  image: { type: String, required: true },
  products: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Product' }],
  sortOrder: { type: Number, default: 0 },
  isActive: { type: Boolean, default: true }
}, { timestamps: true });

// Max 4 products per banner
bannerSchema.pre('save', function (next) {
  if (this.products && this.products.length > 4) {
    this.products = this.products.slice(0, 4);
  }
  next();
});

module.exports = mongoose.model('Banner', bannerSchema);

File: flutter_admin_kapda_junction/lib/domain/entities/banner.dart
import 'package:equatable/equatable.dart';

class BannerProduct extends Equatable {
  final String id;
  final String name;
  final double price;
  final String? image;

  const BannerProduct({required this.id, required this.name, required this.price, this.image});

  @override
  List<Object?> get props => [id];
}

class AppBanner extends Equatable {
  final String id;
  final String image;
  final List<BannerProduct> products;
  final int sortOrder;
  final bool isActive;

  const AppBanner({
    required this.id, required this.image, required this.products,
    required this.sortOrder, required this.isActive,
  });

  @override
  List<Object?> get props => [id];
}
